src/datasets/utils/file_utils.py

```diff
@@ -472,7 +498,8 @@ def get_from_cache(
             )
         elif response is not None and response.status_code == 404:
             raise FileNotFoundError("Couldn't find file at {}".format(url))
-        raise ConnectionError("Couldn't reach {}".format(url))
+        elif not url.startswith("ftp"):
+            raise ConnectionError("Couldn't reach {}".format(url))
```

Comment:
At this point it would be nice to throw a `ConnectionError` if ftp login fails, and a `FileNotFoundError` if the file doesn't exist (if the ftp.size() fails for example).

Only those two kinds of errors are handled to provide indications to the user when it crashes.

Reply:
added these errors